conf/modules.config

```diff
@@ -244,13 +244,83 @@ process {
         ]
     }
 
-    withName: SAMTOOLS_FLAGSTAT {
+    withName: SAMTOOLS_FLAGSTAT_MAPPED {
         publishDir = [
             path: { "${params.outdir}/mapping/" },
             mode: params.publish_dir_mode,
             pattern: '*.flagstat'
         ]
     }
 
+    withName: FILTER_BAM_FRAGMENT_LENGTH {
+        publishDir = [
+            path: { "${params.outdir}/bam_filtering/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.filtered.bam',
+            enabled: params.bamfiltering_savefilteredbams
+        ]
+        ext.args = "-l ${params.bamfiltering_minreadlength}"
+        ext.prefix = { "${meta.id}_${meta.library_id}" }
+
+    }
+
+    withName: SAMTOOLS_FASTQ_UNMAPPED {
+        publishDir = [
+            path: { "${params.outdir}/bam_filtering/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.fastq.gz',
+            enabled: params.bamfiltering_generateunmappedfastq
+        ]
+        ext.args = [
+            '-f 4',
+        ].join(' ').trim()
+        ext.prefix = { "${meta.id}_${meta.library_id}_unmapped" }
+    }
+
+    withName: SAMTOOLS_VIEW {
+        publishDir = [
+            path: { "${params.outdir}/bam_filtering/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.bam',
+            enabled: params.bamfiltering_savefilteredbams
+        ]
+        ext.args = [
+            "-q ${params.bamfiltering_mappingquality}",
+            params.bamfiltering_retainunmappedgenomicbam ? '' : '-F 4',
```

Comment:
I wonder if it would be better to add a general `bamfilter_filterflag` parameter that defaults to 4?
So people would set the filterflag to 0 if bamfiltering_retainunmappedgenomicbam, but could do all the other filtering steps as well (I for example filter for paired reads as well) 
`-F ${params.bamfilter_filterflag}`

see https://broadinstitute.github.io/picard/explain-flags.html

Reply:
Do you mean to give users freedom which flag to provide (e.g. they could arbrtraily provide `256`?

Reply:
yes, exactly. I would use 1 to filter out unmerged reads :)

Reply:
I have added it, I'm not :100:  sure on the utility as we are using `-F` which is specifying which reads to _exclude_ , but I'll leave it to power users to decide ;).